### PR TITLE
feat: ssr catalogue networks

### PR DESCRIPTION
**Pull request by connoratrug**
Add networks to ssr catalogue

- move query filter builder from page to composable 
- and add test framework ( vitetest) 
- add networkspage 
- implement paging and orderby for details page tables 
- render side modal data using slot props ( removes dependency  of sidemodal on render )
- update landing page and menu
- add refback filter type

Note: use feature flag to disable networks part ( and todo items) while still using the updated components in the cohort part of the app ( see README for details). 

todo: 
- more styling
- select items to show 
- implement variables page ( list and detail)

### CODE REVIEW

File: apps/nuxt3-ssr/package.json [hunk 1]
```diff
@@ -17,18 +18,25 @@
   "devDependencies": {
     "@nuxt/image-edge": "^1.0.0-27809804.194d2d6",
     "@nuxt/vite-builder": "3.4.3",
+    "@nuxt/test-utils": "^3.4.1",
```
Comment (mswertz) [inline apps/nuxt3-ssr/package.json:21]: please fix to a specific version.

File: apps/nuxt3-ssr/pages/[schema]/ssr-catalogue/variables/[variable]/index.vue [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+}
+
+let tocItems = reactive([
+  { label: "Description", id: "description" },
+  { label: "Harmonization status", id: "harmonization-per-cohort" },
+]);
+</script>
+
+<template>
+  <LayoutsDetailPage>
```
Comment (mswertz) [inline apps/nuxt3-ssr/pages/[schema]/ssr-catalogue/variables/[variable]/index.vue:83]: dit is mooi, helemaal component based layout

File: apps/nuxt3-ssr/components/NetworkCard.vue [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+          </p>
+
+          <p class="text-body-base mt-5 block xl:hidden">
+            {{
+              isShowingMobileMoreText
+                ? network?.description
+                : `${network?.description?.substring(
+                    0,
+                    mobileShowMoreTextLength
+                  )}...`
```
Comment (mswertz) [inline apps/nuxt3-ssr/components/NetworkCard.vue:102]: description probably should be shortened 
image

File: apps/nuxt3-ssr/components/MainNavigation.vue [hunk 4]
```diff
@@ -10,7 +10,7 @@ interface PropType {
 }
 
 const props: PropType = withDefaults(defineProps<PropType>(), {
-  maximumButtonShown: 3,
```
Comment (mswertz) [inline apps/nuxt3-ssr/components/MainNavigation.vue:13]: I am not sure where config lists, but I would menu to be 'cohorts', 'networks', 'variables'

image

File: apps/nuxt3-ssr/components/VariableCard.vue [hunk 5]
```diff
@@ -0,0 +1,42 @@
+<script setup lang="ts">
```
Comment (mswertz) [inline apps/nuxt3-ssr/components/VariableCard.vue:1]: I think we should try to make the variable card so usefull that people will not feel the need to drill down, except for wanting to look at harmonisation details.

Then I suppose we need (format, description, unit and optionally permitted values)

image

Comment (mswertz) [reply apps/nuxt3-ssr/components/VariableCard.vue:1]: Harmonization is still a 'stub' right?

File: apps/nuxt3-ssr/pages/[schema]/ssr-catalogue/variables/[variable]/index.vue [hunk 6]
```diff
@@ -0,0 +1,151 @@
+<script setup lang="ts">
```
Comment (mswertz) [inline apps/nuxt3-ssr/pages/[schema]/ssr-catalogue/variables/[variable]/index.vue:1]: This is nice. I think main power of this card would be to navigate harmonization efforts
image

A challenge will be that users overall will want to have filters on the network/cohorts they would like to include in this view also. For now we can accept that we simply will see all and instead focus on the central harmonization overview to provide such navigation.

**Review by mswertz — APPROVED** [review]
Development goes in the right direction, lovely to see. Lets have a chat to shortlist what I believe is minimally needed to move this into production (I think it is actually not that far away). Some suggestions for future PRs.

File: apps/nuxt3-ssr/components/VariableCard.vue (hunk 5 above)
Comment (connoratrug) [reply apps/nuxt3-ssr/components/VariableCard.vue:1]: yes , its a stub , the whole variable / network section is behind a feature flag

File: apps/nuxt3-ssr/package.json (hunk 1 above)
Comment (connoratrug) [reply apps/nuxt3-ssr/package.json]: done

File: apps/nuxt3-ssr/pages/[schema]/ssr-catalogue/variables/[variable]/index.vue (hunk 6 above)
Comment (connoratrug) [reply apps/nuxt3-ssr/pages/[schema]/ssr-catalogue/variables/[variable]/index.vue:1]: note 100% sure about the userflow / user task here , bug agree to get the setup up and running and then maybe add additional local ( part of page ) filter options , or create dedicated ui to support this task

**Pull request merged**